lib/assert.js

```diff
@@ -118,12 +118,13 @@ function areSimilarRegExps(a, b) {
 // For small buffers it's faster to compare the buffer in a loop. The c++
 // barrier including the Uint8Array operation takes the advantage of the faster
 // binary compare otherwise. The break even point was at about 300 characters.
-function areSimilarTypedArrays(a, b) {
+// NOTE: Floats should only use binary comparison in non strict mode!
```

Comment:
Nit: Will this comment be clear to implementers? Would it be better to remove it and let tests catch any errors introduced by maintainers?

Reply:
Likely

Reply:
Sure, I removed the comment and the tests already include cases for this, so there is nothing to add anymore.